test/common_test/joint_mimic_features.cc

```diff
@@ -0,0 +1,417 @@
+
+      // Update previous positions.
+      prismaticJointPrevPos = leaderJoint->GetPosition(0);
+      revoluteJointPrevPos = revoluteFollowerJoint1->GetPosition(0);
+    }
+
+    auto testMimicFcn = [&](double multiplier, double offset, double reference)
+      {
+        // Set mimic joint constraints.
+        // Leader --> Follower
+        // prismatic_joint_1 -> prismatic_joint_2
+        // prismatic_joint_1 -> revolute_joint_1
+        // revolute_joint_1 --> revolute_joint_2
+        // revolute_joint_1 --> prismatic_joint_3
+        prismaticFollowerJoint1->SetMimicConstraint("prismatic_joint_1", "axis", multiplier,
+            offset, reference);
+        revoluteFollowerJoint1->SetMimicConstraint("prismatic_joint_1", "axis" , multiplier,
+            offset, reference);
+        revoluteFollowerJoint2->SetMimicConstraint("revolute_joint_1", "axis", multiplier,
+            offset, reference);
+        prismaticFollowerJoint2->SetMimicConstraint("revolute_joint_1", "axis", multiplier,
+            offset, reference);
+
+        // Reset positions and run a few iterations so the positions reach nontrivial values.
+        leaderJoint->SetPosition(0, 0);
+        prismaticFollowerJoint1->SetPosition(0, 0);
+        prismaticFollowerJoint2->SetPosition(0, 0);
+        revoluteFollowerJoint1->SetPosition(0, 0);
+        revoluteFollowerJoint2->SetPosition(0, 0);
+        for (int _ = 0; _ < 10; _++)
+          world->Step(output, state, input);
+
+        // Follower joint's position should be equal to that of leader joint in previous timestep,
+        // considering the offsets and multipliers.
+        prismaticJointPrevPos = leaderJoint->GetPosition(0);
+        revoluteJointPrevPos = revoluteFollowerJoint1->GetPosition(0);
+        for (int _ = 0; _ < 10; _++)
+        {
+          world->Step(output, state, input);
+          // Check for prismatic -> prismatic mimicking.
+          EXPECT_FLOAT_EQ(multiplier * (prismaticJointPrevPos - reference) + offset,
+              prismaticFollowerJoint1->GetPosition(0));
+          // Check for prismatic -> revolute mimicking.
+          EXPECT_FLOAT_EQ(multiplier * (prismaticJointPrevPos - reference) + offset,
+              revoluteFollowerJoint1->GetPosition(0));
+          // Check for revolute -> revolute mimicking.
+          EXPECT_FLOAT_EQ(multiplier * (revoluteJointPrevPos - reference) + offset,
+              revoluteFollowerJoint2->GetPosition(0));
+          // Check for revolute --> prismatic mimicking.
+          EXPECT_FLOAT_EQ(multiplier * (revoluteJointPrevPos - reference) + offset,
+              prismaticFollowerJoint2->GetPosition(0));
+
+          // Update previous positions.
+          prismaticJointPrevPos = leaderJoint->GetPosition(0);
+          revoluteJointPrevPos = revoluteFollowerJoint1->GetPosition(0);
+        }
+      };
+
+    // Testing with different (multiplier, offset, reference) combinations.
+    testMimicFcn(1, 0, 0);
+    testMimicFcn(-1, 0, 0);
```

Comment:
calling `testMimicFcn` sequentially results in multiple calls to `SetMimicConstraint` for a given joint, which appears to work with the dartsim implementation in this PR, but it causes issues with the bullet-featherstone implementation in #517 that adds a new constraint for each call to `SetMimicConstraint`. Probably I need to update #517 to clear any existing mimic constraints when `SetMimicConstraint` is called. We should document in the `Joint.hh` header what the expected behavior is for multiple subsequent calls

Reply:
I've updated https://github.com/gazebosim/gz-physics/pull/517 to clear existing mimic constraints. I made some adjustments to the test expectations, but I've resolved the concern in this comment